### PR TITLE
#170514783-activate/deactivate accommodation 

### PR DESCRIPTION
**Pull request by ngireric123**
#### What does this PR do?
- activate/deactivate accommodation
#### Description of Task to be completed?
- All  travel admins should activate or deactivate accommodation
#### How should this be manually tested?
- Clone this branch
- run npm i
- run npm run dev   
- in your browser navigate to localhost:<:port>/deactivated
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#170514783
#### Screenshots (if appropriate)
image

#### Questions:
N/A

### CODE REVIEW

**Review by Cheza-Dzabala — CHANGES_REQUESTED** [review]
Greetings @ngireric123 Seen the app. Good job, however, the feature is not working as it should. All accommodations say "Deactivate Accommodation" even when the accommodation is already deactivated and is awaiting activation. Secondly, no success alerts are displayed upon activation/deactivation. 

I suggest redirection to a single accommodation page when the accommodation is activated.

File: src/components/pages/SingleAccommodation.js [hunk 1]
```diff
@@ -301,6 +302,25 @@ export class SingleAccommodation extends React.Component {
                         Rooms available
                           </Badge>
                     </span>
+
```
Comment (sabin18) [inline src/components/pages/SingleAccommodation.js]: remove these comments.

File: src/tests/accommodationTests/DeactivatedListItemTest.spec.js [hunk 2]
```diff
@@ -79,5 +80,15 @@ describe('Test AccommodationItem', () => {
   it('should mount ThumbUpOutlinedIcon', () => {
     const component = setUp1({});
      expect(component.find(ThumbUpOutlinedIcon));
+    });
+  it('should mount and render accommodationItem Correctly', () => {
+    const component = setUp({});
+    checkRole.checkHost = jest.fn();
+    checkRole.checkHost.mockReturnValue(false);
+    // const likeButton = findByTestAttribute(component, 'like-button');
```
Comment (sabin18) [inline src/tests/accommodationTests/DeactivatedListItemTest.spec.js]: remove these comments.

**Review by sabin18 — CHANGES_REQUESTED** [review]
good job @ngireric123 , work on the comment I left on this PR and increase test coverage.

**Review by sabin18 — APPROVED** [review]
great job.

**Review by Cheza-Dzabala — APPROVED** [review]

File: src/actions/accommodationActions.js [hunk 3]
```diff
@@ -136,3 +142,18 @@ export const getDeactivatedAccommodation = () => (dispatch) => backendCall.get('
       accommodationType(DEACTIVATED_ACCOMMODATION_ERROR, error.response.data),
     );
   });
+
+export const activatedAccommodation = (slug, reasons) => (dispatch) => backendCall.patch(`/accommodations/activate/${slug}`, { reasons }, { headers: activationHeaders })
```
Comment (nignanthomas) [inline src/actions/accommodationActions.js]: I suggest you rename the action to `activateAccommodation` since it is an action it should be a verb.

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Well done @ngireric123 
However I noticed, the following you need to fix:

- When logged in as a requester I should not view `Deactivate Accommodation` button.
Only `Travel Admin` should view it, right?
Screen Shot 2020-01-15 at 14 54 37

- Kindly add the `Link` to deactivate in the Single accommodation as well. I don't wanna have to go back to the `All accommodations to deactivate`

- When accessing the `/accommodations/deactivate` as a requester it breaks.
Redirect me to `all accommodations` when I am not a travel admin.
Screen Shot 2020-01-15 at 15 14 15

Lemme know as soon as it is fixed. Thanks
**PS**: Fix the failing `codeclimate check`, and I did also leave a comment on the action ☝️

**Review by nignanthomas — APPROVED** [review]

**Pull request merged**